src/Octoshift/HttpDownloadService.cs

```diff
@@ -29,13 +28,14 @@ public virtual async Task DownloadToFile(string url, string file)
         {
             _log.LogVerbose($"HTTP GET: {url}");
 
-            using var response = await _httpClient.GetAsync(url);
+            using var response = await _httpClient.GetAsync(url, HttpCompletionOption.ResponseHeadersRead);
             _log.LogVerbose($"RESPONSE ({response.StatusCode}): <truncated>");
 
             response.EnsureSuccessStatusCode();
 
-            var contents = await response.Content.ReadAsStringAsync();
-            await WriteToFile(file, contents);
+            await using var streamToReadFrom = await response.Content.ReadAsStreamAsync();
+            await using var streamToWriteTo = _fileSystemProvider.OpenRead(file);
```

Comment:
I guess you confused this with my other comment in `MigrateRepoCommandHanlder`,  this was good the way it was before, we need to write to the stream of course 😃:
```suggestion
            using var streamToWriteTo = _fileSystemProvider.Open(file, FileMode.Create); 
```

Reply:
Fixed in 7d436e04071bf868990e1d32d6276668c1db1a35 :tada: